app/gosqs/send_message.go

```diff
@@ -0,0 +1,133 @@
+	messageAttributes := requestBody.MessageAttributes
+
+	queueUrl := getQueueFromPath(requestBody.QueueUrl, req.URL.String())
+
+	queueName := ""
+	if queueUrl == "" {
+		vars := mux.Vars(req)
+		queueName = vars["queueName"]
+	} else {
+		uriSegments := strings.Split(queueUrl, "/")
+		queueName = uriSegments[len(uriSegments)-1]
+	}
+
+	if _, ok := app.SyncQueues.Queues[queueName]; !ok {
+		// Queue does not exist
+		return createErrorResponseV1("QueueNotFound")
+	}
+
+	if app.SyncQueues.Queues[queueName].MaximumMessageSize > 0 &&
+		len(messageBody) > app.SyncQueues.Queues[queueName].MaximumMessageSize {
+		// Message size is too big
+		return createErrorResponseV1("MessageTooBig")
+	}
+
+	delaySecs := app.SyncQueues.Queues[queueName].DelaySeconds
+	if mv := req.FormValue("DelaySeconds"); mv != "" {
+		delaySecs, _ = strconv.Atoi(mv)
+	}
+
+	log.Println("Putting Message in Queue:", queueName)
+	msg := app.Message{MessageBody: []byte(messageBody)}
+	if len(messageAttributes) > 0 {
+		oldStyleMessageAttributes := convertToOldMessageAttributeValueStructure(messageAttributes)
+		msg.MessageAttributes = oldStyleMessageAttributes
+		msg.MD5OfMessageAttributes = common.HashAttributes(oldStyleMessageAttributes)
+	}
+	msg.MD5OfMessageBody = common.GetMD5Hash(messageBody)
+	msg.Uuid, _ = common.NewUUID()
+	msg.GroupID = messageGroupID
+	msg.DeduplicationID = messageDeduplicationID
+	msg.SentTime = time.Now()
+	msg.DelaySecs = delaySecs
+
+	app.SyncQueues.Lock()
+	fifoSeqNumber := ""
+	if app.SyncQueues.Queues[queueName].IsFIFO {
+		fifoSeqNumber = app.SyncQueues.Queues[queueName].NextSequenceNumber(messageGroupID)
+	}
+
+	if !app.SyncQueues.Queues[queueName].IsDuplicate(messageDeduplicationID) {
+		app.SyncQueues.Queues[queueName].Messages = append(app.SyncQueues.Queues[queueName].Messages, msg)
+	} else {
+		log.Debugf("Message with deduplicationId [%s] in queue [%s] is duplicate ", messageDeduplicationID, queueName)
+	}
+
+	app.SyncQueues.Queues[queueName].InitDuplicatation(messageDeduplicationID)
+	app.SyncQueues.Unlock()
+	log.Infof("%s: Queue: %s, Message: %s\n", time.Now().Format("2006-01-02 15:04:05"), queueName, msg.MessageBody)
+
+	respStruct := models.SendMessageResponse{
+		Xmlns: "http://queue.amazonaws.com/doc/2012-11-05/",
+		Result: models.SendMessageResult{
+			MD5OfMessageAttributes: msg.MD5OfMessageAttributes,
+			MD5OfMessageBody:       msg.MD5OfMessageBody,
+			MessageId:              msg.Uuid,
+			SequenceNumber:         fifoSeqNumber,
+		},
+		Metadata: app.ResponseMetadata{
+			RequestId: "00000000-0000-0000-0000-000000000000",
+		},
+	}
+
+	return http.StatusOK, respStruct
+}
+
+// TODO:
+// Refactor internal MessageAttribute model between SendMessage and ReceiveMessage
+// from app.MessageAttributeValue(old) to models.MessageAttributeValue(new) and remove this temporary function.
+func convertToOldMessageAttributeValueStructure(newValues map[string]models.MessageAttributeValue) map[string]app.MessageAttributeValue {
```

Comment:
Current internal MessageAttribute model `app.MessageAttributeValue` is a bit difference with recent [official MessageAtteributeValue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_MessageAttributeValue.html).

If there is any matter to use the current struct after migration all the SQS/SNS API, we can refactor the `app.MessageAttributeValue` to latest `models.MessageAttributeValue`.